libsolidity/ast/Types.h

```diff
@@ -860,13 +866,13 @@ class ArraySliceType: public ReferenceType
 	std::string toString(bool _short) const override;
 	TypePointer mobileType() const override;
 
-	BoolResult validForLocation(DataLocation _loc) const override { return m_arrayType.validForLocation(_loc); }
-
 	ArrayType const& arrayType() const { return m_arrayType; }
 	u256 memoryDataSize() const override { solAssert(false, ""); }
 
 	std::unique_ptr<ReferenceType> copyForLocation(DataLocation, bool) const override { solAssert(false, ""); }
 
+	std::vector<Type const*> subtypes() const override { return { &m_arrayType }; }
```

Comment:
https://github.com/ethereum/solidity/pull/9746#discussion_r484523296
```suggestion
	std::vector<Type const*> subtypes() const override { return {&m_arrayType}; }
```

Reply:
Done.